### PR TITLE
[Docs] Fix doc typos in MMDetection 3.x

### PR DESCRIPTION
**Pull request by RangeKing**
## Modification

Fix typos in
|doc name | en version | zh_cn version |
|-|:-:|:-:|
|get_started.md|√| √|
|config.md        |√| √|
|train.md           |√|   |
|inference.md   |√|   |
|new_model.md|√|   |
|finetume.md   |√|   |
|dataset_prepare.md|√|   |

### CODE REVIEW

**Comment by BIGWangYuDong** [comment]
Right now, all docs LGTM

File: docs/en/user_guides/train.md [hunk 1]
```diff
@@ -8,19 +8,19 @@ This section will show how to train _predefined_ models (under [configs](https:/
 Training requires preparing datasets too. See section [Prepare datasets](#prepare-datasets) above for details.
 
 **Note**:
-Currently, the config files under `configs/cityscapes` use COCO pretrained weights to initialize.
+Currently, the config files under `configs/cityscapes` use COCO pre-trained weights to initialize.
```
Comment (zytx121) [inline docs/en/user_guides/train.md:11]:
```suggestion
Currently, the config files under `configs/cityscapes` use COCO pretrained weights to initialize.
```

File: docs/en/user_guides/config.md [hunk 2]
```diff
@@ -24,13 +24,13 @@ model = dict(
...
-        out_indices=(0, 1, 2, 3),  # The index of output feature maps produced in each stages
+        out_indices=(0, 1, 2, 3),  # The index of output feature maps produced in each stage
         frozen_stages=1,  # The weights in the first stage are frozen
         norm_cfg=dict(  # The config of normalization layers.
             type='BN',  # Type of norm layer, usually it is BN or GN
             requires_grad=True),  # Whether to train the gamma and beta in BN
         norm_eval=True,  # Whether to freeze the statistics in BN
-        style='pytorch', # The style of backbone, 'pytorch' means that stride 2 layers are in 3x3 conv, 'caffe' means stride 2 layers are in 1x1 convs.
+        style='pytorch', # The style of backbone, 'pytorch' means that stride 2 layers are in 3x3 Conv, 'caffe' means stride 2 layers are in 1x1 Convs.
     	init_cfg=dict(type='Pretrained', checkpoint='torchvision://resnet50')),  # The ImageNet pretrained backbone to be loaded
```
Comment (zytx121) [inline docs/en/user_guides/config.md:34]:
```suggestion
    	init_cfg=dict(type='Pretrained', checkpoint='torchvision://resnet50')),  # The ImageNet pre-trained backbone to be loaded
```

**Comment by ZwwWayne** [comment]
Thanks for the great PR. The current modification is suitable for merge. So this PR will be merged first. Feel free to create a new PR if you want to further modify the doc.

**Pull request merged**